### ISSUE TITLE
samba shares can send invalid "0" change flag events and trip a wxFAIL in int wxIOCPThread::Native2WatcherFlags

### ISSUE THREAD

**Issue by wxtrac** [wxWidgets/wxWidgets#18953]
*Issue migrated from trac ticket # 18953*

**component:** wxMSW | **priority:** normal | **resolution:** fixed

#### 2020-10-29 02:48:13: mroszko (MARK ROSZKO) created the issue

___
I have a few circumstances now where I have narrowed down with users with samba network shares and wxFileSystemWatcher.

In Windows land, the event generation is entirely up to the underlying fs. So if samba is behind the network share, it's free to generate as many bad events as it wants and Windows will simply pass it on :/

This line here
https://github.com/wxWidgets/wxWidgets/blob/master/src/msw/fswatcher.cpp#L396

`
    wxFAIL_MSG(wxString::Format("Unknown file notify change %u", flags));
`

is being hit as a result.

While the fault most likely lies within samba itself. It seems a wxFAIL here is far too extreme and just passively ignoring the event would be much better.

**Comment by sethhillbrand** [comment]
Can we re-open this issue?  The fix here catches some but not all of the invalid flags that can be sent.  And the wxFAIL still causes issues.  Is there a reason that we cannot just log and ignore the invalid flag?  Do we need to FAIL the whole program?

**Comment by vadz** [comment]
No, data coming from outside of the program definitely shouldn't trigger an assert failure, this should be at most `wxLogDebug()` and maybe just silently ignored.

I'll change this, but to help to choose between the last 2 options, do you know if this happens just episodically (in which case we probably should log it to help diagnose a potential problem) or can it happen continuously?

**Comment by sethhillbrand** [comment]
With a poorly configured samba server, it can happen repeatedly.

**Comment by vadz** [comment]
OK, thanks, I've created #25201 which contains 380cf411bc7c1e5b3ec82d01656d3ec61e50bf49 fixing this and will backport this commit to 3.2 too.

**Comment by sethhillbrand** [comment]
Thanks Vadim!  Cc: @marekr
